staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1101,17 +1115,19 @@ func (c *Cacher) LastSyncResourceVersion() (uint64, error) {
 
 // cacherListerWatcher opaques storage.Interface to expose cache.ListerWatcher.
 type cacherListerWatcher struct {
-	storage        storage.Interface
-	resourcePrefix string
-	newListFunc    func() runtime.Object
+	storage                storage.Interface
+	resourcePrefix         string
+	newListFunc            func() runtime.Object
+	kcpClusterAwareContext context.Context
 }
 
 // NewCacherListerWatcher returns a storage.Interface backed ListerWatcher.
-func NewCacherListerWatcher(storage storage.Interface, resourcePrefix string, newListFunc func() runtime.Object) cache.ListerWatcher {
+func NewCacherListerWatcher(storage storage.Interface, resourcePrefix string, newListFunc func() runtime.Object, kcpExtraStorageMetadata *storagebackend.KcpStorageMetadata) cache.ListerWatcher {
 	return &cacherListerWatcher{
-		storage:        storage,
-		resourcePrefix: resourcePrefix,
-		newListFunc:    newListFunc,
+		storage:                storage,
+		resourcePrefix:         resourcePrefix,
+		newListFunc:            newListFunc,
+		kcpClusterAwareContext: createKCPClusterAwareContext(kcpExtraStorageMetadata),
```

Comment:
storing a context still feels strange. Why not store `kcpExtraStorageMetadata` and call `createKCPClusterAwareContext` further down?

Reply:
storing a ctx seems easier here because I need to create the context only once and reuse it multiple times